### PR TITLE
Update contracts and AluScript

### PR DESCRIPTION
**Pull request by crisdut**
**Depends**

- https://github.com/RGB-WG/rgb-std/pull/152
- https://github.com/RGB-WG/rgb-core/pull/201

### CODE REVIEW

File: src/uda.rs [hunk 1]
```diff
@@ -48,31 +47,31 @@ pub fn uda_schema() -> SubSchema {
...
+        ldg     0x0836,a8[0],s16[1]         ;
         // Put 0 to a16[0]
         put     a16[0],0x00                 ;
         // Extract 128 bits from the beginning of s[1] into r128[1]
         extr    s16[1],r128[1],a16[0]       ;
         // a32[0] now has token index from global state
         spy     a32[0],r128[1]              ;
         // Read owned state into s[1]
-        lds     0x0FA0,0,s16[1]             ;
+        lds     0x0FA0,a16[0],s16[1]        ;
```
[the remark this answers is not in the document]
Comment (crisdut) [reply src/uda.rs:60]: Great, I will change it.

File: src/uda.rs [hunk 2]
```diff
@@ -48,31 +47,31 @@ pub fn uda_schema() -> SubSchema {
...
         // Extract 128 bits from the beginning of s[1] into r128[1]
         extr    s16[1],r128[1],a16[0]       ;
         // a32[1] now has token index from allocation
         spy     a32[1],r128[1]              ;
         // check that token indexes match
         eq.n    a32[0],a32[1]               ;
         // if they do, jump to the next check
-        jif     0x0028                      ;
+        jif     0x0029                      ;
```
[the remark this answers is not in the document]
Comment (crisdut) [reply src/uda.rs:68]: Humm, I tested it, but I will review again.

File: Cargo.toml [hunk 3]
```diff
@@ -31,5 +31,6 @@ serde_yaml = "0.9.27"
 all=[]
 
 [patch.crates-io]
-rgb-core = { git = "https://github.com/RGB-WG/rgb-core", branch = "master" }
-rgb-std = { git = "https://github.com/RGB-WG/rgb-std", branch = "master" }
+# TODO: Remove after merge
+rgb-core = { git = "https://github.com/crisdut/rgb-core", branch = "feat/udc" }
+rgb-std = { git = "https://github.com/crisdut/rgb-std", branch = "fix/input-size" }
```
Comment (dr-orlovsky) [inline Cargo.toml:36]: Now we have the upstream PRs merged and this can be removed

**Comment by crisdut** [comment]
Hi @dr-orlovsky,

I update the dependencies and UDA script to avoid registers re-use.

**Pull request merged**